pallets/creator-staking/src/inflation.rs

```diff
@@ -0,0 +1,44 @@
+use crate::*;
+use frame_support::traits::{Currency, Get, OnTimestampSet, Imbalance};
+use sp_runtime::traits::{Saturating, SaturatedConversion, UniqueSaturatedInto, Zero};
+
+type NegativeImbalanceOf<T> = <<T as Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::NegativeImbalance;
+
+impl<Moment, T: Config> OnTimestampSet<Moment> for Pallet<T> {
+    fn on_timestamp_set(_moment: Moment) {
+        let total_issuance = T::Currency::total_issuance();
+        let reward_amount: BalanceOf<T> = T::CurrentAnnualInflation::get() * total_issuance
+            / T::BlocksPerYear::get().saturated_into::<u32>().unique_saturated_into();
+
+        let inflation = T::Currency::issue(reward_amount);
+        Self::distribute_rewards(inflation);
+    }
+}
+
+impl<T: Config> Pallet<T> {
+    fn distribute_rewards(block_reward: NegativeImbalanceOf<T>) {
+        let distro_params = Self::reward_config();
+
+        // Pre-calculate balance which will be deposited for each beneficiary
+        let stakers_balance = distro_params.stakers_percent * block_reward.peek();
+        let creators_balance = distro_params.creators_percent * block_reward.peek();
+
+        // Prepare imbalances
+        let (creators_imbalance, remainder) = block_reward.split(creators_balance);
+        let (stakers_imbalance, treasury_imbalance) = remainder.split(stakers_balance);
```

Comment:
Why we cannot do this simpler? for example:
```rust
let treasury_imbalance = block_reward.sub(creators_balance).sub(stakers_balance);
```

Reply:
Resolve creating requires `NegativeImbalance` to be passed as `value`.
To avoid converting to `Balance` and then back again to `NegativeImbalance`, we do operate with the last one using accessible functions.

Furthermore, there may some crumbs left after distribution, and `split` will return us that amount, so we can consider it.